test/parallel/test-assert-objects.js

```diff
@@ -97,6 +97,21 @@ describe('Object Comparison Tests', () => {
           actual: [1, 'two', true],
           expected: [1, 'two', false],
         },
+        {
+          description: 'throws when comparing [0] with [-0]',
+          actual: [0],
+          expected: [-0],
+        },
+        {
+          description: 'throws when comparing [0, 0, 0] with [0, -0]',
+          actual: [0, 0, 0],
+          expected: [0, -0],
+        },
+        {
+          description: 'throws when comparing [-0] with [0]',
+          actual: [0],
+          expected: [-0],
+        },
```

Comment:
```suggestion
        {
          description: 'throws when comparing [0, 0, 0] with [0, -0]',
          actual: [0, 0, 0],
          expected: [0, -0],
        },
        {
          description: 'throws when comparing [0, -0, 0] with [0, 0]',
          actual: [0, -0, 0],
          expected: [0, 0],
        },
        {
          description: 'throws when comparing [-0] with [0]',
          actual: [-0],
          expected: [0],
        },
```

Reply:
@LiviaMedeiros I added more tests!  
There is a problem with one of yours tho:

```js
{
     description: 'throws when comparing [0, -0, 0] with [0, 0]',
     actual: [0, -0, 0],
     expected: [0, 0],
},
```

this should not throw, because bot the element in the `expected` array are present in the `actual` one